bindings/java/src/org/sleuthkit/datamodel/AbstractFile.java

```diff
@@ -908,6 +908,26 @@ public boolean isDir() {
 		return (metaType.equals(TSK_FS_META_TYPE_ENUM.TSK_FS_META_TYPE_DIR)
 				|| metaType.equals(TSK_FS_META_TYPE_ENUM.TSK_FS_META_TYPE_VIRT_DIR));
 	}
+	
+	/**
+	 * Count of Abstract File's children of specified types. The types are
+	 * instances of TskData.TSK_FS_NAME_TYPE_ENUM types and the matching is
+	 * performed against the tsk_files.dir_type column. Some usage examples are
+	 * to get a count of all subdirectories, which requires searching for all
+	 * children of types TskData.TSK_FS_NAME_TYPE_ENUM.DIR "OR"
+	 * TskData.TSK_FS_NAME_TYPE_ENUM.VIRT_DIR.
+	 *
+	 * @param types List of TskData.TSK_FS_NAME_TYPE_ENUM types.
+	 *
+	 * @return Total count of children of the specified types.
+	 *
+	 * @throws org.sleuthkit.datamodel.TskCoreException exception thrown if a
+	 *                                                  critical error occurs
+	 *                                                  within tsk core
+	 */
+	public int countChildrenOfType(List<TSK_FS_NAME_TYPE_ENUM> types) throws TskCoreException {
```

Comment:
Does this need to be marked as `@Beta` by any chance?

Reply:
No idea when we apply this marking. @APriestman ?

Reply:
I've been putting it on most of my new stuff just so we're not committing to all the new APIs :)

Reply:
How non-committal of us! ;) 

Done.